go-controller/pkg/ovn/libovsdbops/acl.go

```diff
@@ -0,0 +1,192 @@
+	}).List(&acls)
+	if err != nil {
+		return "", fmt.Errorf("can't find ACL by equivalence %+v: %v", *acl, err)
+	}
+
+	if len(acls) > 1 {
+		return "", fmt.Errorf("unexpectedly found multiple equivalent ACLs: %+v", acls)
+	}
+
+	if len(acls) == 1 {
+		return acls[0].UUID, nil
+	}
+
+	return "", nil
+}
+
+func BuildACL(name, direction nbdb.ACLDirection, priority int, match string, action nbdb.ACLAction, meter string, severity nbdb.ACLSeverity, log bool, externalIds map[string]string) *nbdb.ACL {
+	uuid := BuildNamedUUID()
+	name = fmt.Sprintf("%.63s", name)
+	return &nbdb.ACL{
+		UUID:        uuid,
+		Name:        &name,
+		Direction:   direction,
+		Match:       match,
+		Action:      action,
+		Priority:    priority,
+		Severity:    &severity,
+		Log:         log,
+		Meter:       &meter,
+		ExternalIDs: externalIds,
+	}
+}
+
+func createOrUpdateACLOps(nbClient libovsdbclient.Client, ops []libovsdb.Operation, acl *nbdb.ACL) ([]libovsdb.Operation, error) {
+	if ops == nil {
+		ops = []libovsdb.Operation{}
+	}
+
+	uuid, err := findACL(nbClient, acl)
+	if err != nil {
+		return nil, err
+	}
+
+	// If ACL already exists, update it
+	if uuid != "" {
+		acl.UUID = ""
+		op, err := nbClient.Where(&nbdb.ACL{UUID: uuid}).Update(acl)
+		if err != nil {
+			return nil, err
+		}
+		ops = append(ops, op...)
+		acl.UUID = uuid
+		return ops, nil
+	}
+
+	op, err := nbClient.Create(acl)
+	if err != nil {
+		return nil, err
+	}
+	ops = append(ops, op...)
+
+	return ops, nil
+}
+
+func CreateOrUpdateACLsOps(nbClient libovsdbclient.Client, ops []libovsdb.Operation, acls ...*nbdb.ACL) ([]libovsdb.Operation, error) {
+	if ops == nil {
+		ops = []libovsdb.Operation{}
+	}
+	for _, acl := range acls {
+		var err error
+		ops, err = createOrUpdateACLOps(nbClient, ops, acl)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return ops, nil
+}
+
+func CreateOrUpdateACLs(nbClient libovsdbclient.Client, acls ...*nbdb.ACL) error {
+	ops, err := CreateOrUpdateACLsOps(nbClient, nil, acls...)
+	if err != nil {
+		return err
+	}
+
+	_, err = TransactAndCheck(nbClient, ops)
+	return err
+}
+
+func UpdateACLLoggingOps(nbClient libovsdbclient.Client, ops []libovsdb.Operation, acl *nbdb.ACL) ([]libovsdb.Operation, error) {
+	if ops == nil {
+		ops = []libovsdb.Operation{}
+	}
+
+	uuid, err := findACL(nbClient, acl)
+	if err != nil {
+		return nil, err
+	}
+
+	if uuid == "" {
+		return nil, fmt.Errorf("error, acl not found %+v", acl)
+	}
+
+	acl.UUID = ""
+	op, err := nbClient.Where(&nbdb.ACL{UUID: uuid}).Update(acl, &acl.Severity, &acl.Log)
+	if err != nil {
+		return nil, err
+	}
+	ops = append(ops, op...)
+	acl.UUID = uuid
+
+	return ops, nil
+}
+
+func UpdateACLLogging(nbClient libovsdbclient.Client, acl *nbdb.ACL) error {
+	ops, err := UpdateACLLoggingOps(nbClient, nil, acl)
+	if err != nil {
+		return err
+	}
+
+	_, err = TransactAndCheck(nbClient, ops)
+	return err
+}
+
+func FindRejectACLs(nbClient libovsdbclient.Client) ([]*nbdb.ACL, error) {
```

Comment:
If you want to just move the reject ACL cleanup code to the service controller, feel free - that is an edge case that will go away soon.